### PR TITLE
Fix auto mtls for PASSTHROUGH load balancer mode

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/23494

Note: this is stacked on https://github.com/istio/istio/pull/31285. I will rebase it once that merges

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -1822,14 +1822,17 @@ func (ps *PushContext) NetworkGatewaysByNetwork(network string) []*Gateway {
...
 		return MTLSUnknown
 	}
 
-	// 1. Check service instances' tls mode, mainly used for headless service.
-	if service.Resolution == Passthrough {
+	// For passthrough traffic (headless service or explicitly defined in DestinationRule), we look at the instances
+	// If ALL instances have a sidecar, we enable TLS, otherwise we disable
+	// TODO(https://github.com/istio/istio/issues/27376) enable mixed deployments
+	// A service with passthrough resolution is always passthrough, regardless of the TrafficPolicy.
+	if service.Resolution == Passthrough || tp.GetLoadBalancer().GetSimple() == networking.LoadBalancerSettings_PASSTHROUGH {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context.go:1835]: One confusion is: when service is not passthrough/headless, how comes it selects the cluster?

I understand it possibly goes through the passthrough cluster.

Comment (howardjohn) [reply pilot/pkg/model/push_context.go:1835]: you can define passthrough destination rule and still send to the service Ip which would match tcp, or include host header and match http.

If you see the original issue shows knative plan to use this

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go:1835]: I understand the use case now.

**Review by hzxuzhonghu — APPROVED** [review]

File: releasenotes/notes/auto-mtls-passthrough.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: networking
+issue:
+- 23494
+releaseNotes:
+- |
+  **Fixed** an issue services with `PASSTHROUGH` load balancing were always sent mTLS traffic, even if the destinations did not support mTLS.
```
Comment (ericvn) [inline releasenotes/notes/auto-mtls-passthrough.yaml:8]: an issue **where** services

**Review by incfly — APPROVED** [review]

File: releasenotes/notes/auto-mtls-passthrough.yaml (hunk 2 above)
Comment (howardjohn) [reply releasenotes/notes/auto-mtls-passthrough.yaml]: done

**Review by jacob-delgado — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

**Pull request merged**